src/jit/decomposelongs.cpp

```diff
@@ -1380,9 +1366,30 @@ GenTree* DecomposeLongs::DecomposeRotate(LIR::Use& use)
     }
     else
     {
-        Range().Remove(gtLong);
-        loOp1 = RepresentOpAsLocalVar(loOp1, gtLong, &gtLong->gtOp.gtOp1);
-        hiOp1 = RepresentOpAsLocalVar(hiOp1, gtLong, &gtLong->gtOp.gtOp2);
+        GenTree* loResult;
+        GenTree* hiResult;
+
+        if (count > 32)
+        {
+            // If count > 32, we swap hi and lo, and subtract 32 from count
+            hiOp1 = gtLong->gtGetOp1();
+            loOp1 = gtLong->gtGetOp2();
```

Comment:
They aren't quite repeated. In the > 32 chase, hiOp is gtLong's op1 and loOp is gtLong's op2